vochain/processarchive/processarchive.go

```diff
@@ -0,0 +1,140 @@
+package processarchive
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"go.vocdoni.io/dvote/data"
+	"go.vocdoni.io/dvote/log"
+	"go.vocdoni.io/dvote/types"
+	"go.vocdoni.io/dvote/vochain"
+	"go.vocdoni.io/proto/build/go/models"
+)
+
+type ProcessArchive struct {
+	vochain     *vochain.BaseApplication
+	ipfs        *data.IPFSHandle
+	storage     *jsonStorage
+	pprocs      []*Process
+	publish     chan (bool)
+	lastUpdate  time.Time
+	close       chan (bool)
+	publishLock sync.Mutex
+}
+
+type Process struct {
+	Process *models.Process
+	Votes   uint32
+}
+
+type jsonStorage struct {
+	datadir string
+	lock    sync.RWMutex
+}
+
+func NewJsonStorage(datadir string) (*jsonStorage, error) {
+	err := os.MkdirAll(datadir, 0o750)
+	if err != nil {
+		return nil, err
+	}
+	return &jsonStorage{datadir: datadir}, nil
+}
+
+func (js *jsonStorage) AddProcess(p *Process) error {
+	if p == nil || p.Process == nil || len(p.Process.ProcessId) != types.ProcessIDsize {
+		return fmt.Errorf("process not valid")
+	}
+	data, err := json.Marshal(p)
+	if err != nil {
+		return err
+	}
+	js.lock.Lock()
+	defer js.lock.Unlock()
+	return os.WriteFile(filepath.Join(js.datadir, fmt.Sprintf("%x", p.Process.ProcessId)), data, 0o644)
```

Comment:
atomically writing files is possible with https://github.com/google/renameio, in which case you can get rid of the global locking.

Reply:
TO-DO